build.cmd

```diff
@@ -4,51 +4,96 @@ PUSHD %~dp0
 SETLOCAL
 SETLOCAL ENABLEDELAYEDEXPANSION
 
+:Loop
+if [%1]==[] GOTO Begin
+
+if /I [%1]==[Debug] (
+    SET Configuration=%1
+    GOTO Next
+)
+
+if /I [%1]==[Release] (
+    SET Configuration=%1
+    GOTO Next
+)
+
+if /I [%1]==[PROD] (
+    SET Environment=PROD
+    GOTO Next
+)
+
+if /I [%1]==[raw] (
+    SET UpdateTemplate=%1
+    GOTO Next
+)
+if /I [%1]==[template] (
+    SET UpdateTemplate=%1
+    GOTO Next
+)
+
+:Next
+SHIFT /1
+GOTO Loop
+
+:Begin
 IF NOT DEFINED VisualStudioVersion (
     IF DEFINED VS140COMNTOOLS (
         CALL "%VS140COMNTOOLS%\VsDevCmd.bat"
-        GOTO :EnvSet
+        GOTO EnvSet
     )
 
     ECHO Error: build.cmd requires Visual Studio 2015.
     SET ERRORLEVEL=1
-    GOTO :Exit
+    GOTO Exit
 )
 
 :EnvSet
 SET BuildProj=%~dp0All.sln
-SET Configuration=%1
-IF '%Configuration%'=='' (
+IF [%Configuration%]==[] (
     SET Configuration=Release
 )
-SET Environment=%2
-IF '%Environment%'=='PROD' (
+IF /I [%Environment%]==[PROD] (
     ECHO Updating version for PROD environment
     CALL UpdateVersion.cmd
 
-    IF NOT '!ERRORLEVEL!'=='0' (
+    IF NOT [!ERRORLEVEL!]==[0] (
         ECHO ERROR: Error occurs when updating version
-        GOTO :Exit
+        GOTO Exit
     )
 )
-
-:: Check if DNU exists globally
-:: DNU is OPTIONAL
-WHERE dnu >NUL
-IF NOT '%ERRORLEVEL%'=='0' (
-    ECHO ERROR: DNU is not successfully configured.
-    ECHO ERROR: Please follow http://docs.asp.net/en/latest/getting-started/installing-on-windows.html#install-the-net-version-manager-dnvm to install dnvm.
-    ECHO ERROR: If dnvm is installed, use `dnvm list` to show available dnx runtime, and use `dnvm use` to select the default dnx runtime
-    GOTO :Exit
+IF /I [%UpdateTemplate%]==[template] (
+    CALL UpdateTemplate.cmd
```

Comment:
in this way, it not only updates template but also build project.

---
In reply to: [68713565](https://github.com/dotnet/docfx/pull/435#discussion_r68713565) [](ancestors = 68713565)